public/app/features/alerting/unified/components/backtesting/BacktestPanel.tsx

```diff
@@ -0,0 +1,327 @@
+  const [instancesFilter, setInstancesFilter] = useState('');
+  const [showRefreshBanner, setShowRefreshBanner] = useState(false);
+  const logsRef = useRef<Map<number, HTMLElement>>(new Map<number, HTMLElement>());
+  const lastRunRuleDefinitionRef = useRef<RuleFormValues | null>(null);
+
+  const { getValues, setValue } = useForm({ defaultValues: { query: '' } });
+
+  const handleRunBacktest = async () => {
+    setIsLoading(true);
+    setError(null);
+    setStateHistory(undefined);
+    setShowRefreshBanner(false);
+    lastRunRuleDefinitionRef.current = ruleDefinition;
+
+    try {
+      console.log(ruleDefinition);
+      // Convert form values to the proper AlertRule format
+      const alertRule = formValuesToRulerGrafanaRuleDTO(ruleDefinition);
+      console.log(alertRule);
+
+      // Build requestBody matching BacktestConfig struct
+      const requestBody = {
+        // Required time range fields
+        from: timeRange.from.toISOString(),
+        to: timeRange.to.toISOString(),
+        interval: ruleDefinition.evaluateEvery,
+
+        // Required alert definition fields
+        condition: alertRule.grafana_alert.condition,
+        data: alertRule.grafana_alert.data,
+        title: alertRule.grafana_alert.title,
+        no_data_state: alertRule.grafana_alert.no_data_state,
+        exec_err_state: alertRule.grafana_alert.exec_err_state,
+
+        // Optional duration fields
+        for: alertRule.for,
+        keep_firing_for: alertRule.keep_firing_for,
+
+        // Optional metadata fields
+        labels: alertRule.labels,
+        missing_series_evals_to_resolve: alertRule.grafana_alert.missing_series_evals_to_resolve,
+
+        // Optional rule identification fields
+        uid: alertRule.grafana_alert.uid,
+        rule_group: ruleDefinition.group,
+        namespace_uid: ruleDefinition.folder?.uid,
+      };
+
+      const response = await fetch('/api/v1/rule/backtest', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(requestBody),
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+
+      const data: DataFrameJSON = await response.json();
+      console.log('Backtesting API response:', data);
+
+      setStateHistory(data);
+    } catch (err) {
+      setError(err instanceof Error ? err : new Error('Unknown error occurred'));
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  // Update time range when initialTimeRange changes
+  useEffect(() => {
+    if (initialTimeRange) {
+      setTimeRange(initialTimeRange);
+    }
+  }, [initialTimeRange]);
+
+  // Trigger backtest when triggerRun changes (for predefined time ranges)
+  useEffect(() => {
+    if (triggerRun && triggerRun > 0) {
+      handleRunBacktest();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Disabling this is suspicious and usually indicates that we're doing something non-idiomatic